js/global.js

```diff
@@ -442,3 +442,13 @@ M.throttle = function(func, wait, options) {
     return result;
   };
 };
+
+/* Feature detection */
+var passiveIfSupported = false;
+try {
+    window.addEventListener("test", null, 
+        Object.defineProperty({}, "passive", {
+            get: function() { passiveIfSupported = { passive: false }; }
```

Comment:
Shouldn't this be true?

Reply:
Possibly - I originally set `passive:true`, but @waeco set `passive:false` (which also works and removes the violation), and ideally I want something that can be shared across `waves.js` as well. (How both true and false can work is beyond me)

https://github.com/materializecss/materialize/commit/3afddefe2cadef09cb0221d9cdbbc61095f27b06

Reply:
Hm, because https://github.com/materializecss/materialize/pull/21/commits/95580a524199ca8774ccff87f37d13911b9626d4 sets it to `true`. @WaeCo

Reply:
False is the default value. I did not want to accidentally break anything, that's why I set it to false.

However according to [MDN addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener)
> passive
A Boolean that, if true, indicates that the function specified by listener will never call preventDefault(). If a passive listener does call preventDefault(), the user agent will do nothing other than generate a console warning. See Improving scrolling performance with passive listeners to learn more.

I can not find any preventDefault() calls in Waves, so I think true is actually correct and should work there too.

Reply:
In that case sidenav needs to be false, because it does call `preventDefault()`. Does this mean we need separate feature defection functions?

Reply:
As far as I can see `preventDefault()` is only used in `handleTriggerClick()` which does not have `passiveIfSupported` set, so we should be good.